client/views/account/AccountOutOfOfficePage/AccountOutOfOfficeForm.tsx

```diff
@@ -0,0 +1,263 @@
+			isEnabled: false,
+			startDateString,
+			startTimeString,
+			endDateString,
+			endTimeString,
+			customMessage: '',
+			roomIds: [],
+		};
+	}
+
+	return {
+		isEnabled: !!receivedFormValues.isEnabled,
+		customMessage: receivedFormValues.customMessage,
+		roomIds: receivedFormValues.roomIds ?? [],
+		startDateString,
+		startTimeString,
+		endDateString,
+		endTimeString,
+	};
+}
+
+function OutOfOfficeForm({
+	receivedOutOfOfficeValues,
+}: {
+	receivedOutOfOfficeValues: IReceivedFormValues;
+}): JSX.Element {
+	const t = useTranslation() as any;
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const { values, handlers, commit, hasUnsavedChanges } = useForm(
+		getInitialFormValues(receivedOutOfOfficeValues) as any,
+	);
+
+	const {
+		isEnabled,
+		roomIds,
+		customMessage,
+		startDateString,
+		startTimeString,
+		endDateString,
+		endTimeString,
+	}: IFormValues = values as any;
+
+	const {
+		handleIsEnabled,
+		handleCustomMessage,
+		handleStartDateString,
+		handleStartTimeString,
+		handleEndDateString,
+		handleEndTimeString,
+		handleRoomIds,
+	} = handlers;
+
+	const toggleOutOfOffice = useEndpointAction(
+		'POST',
+		'outOfOffice.toggle',
+		useMemo(
+			() => ({
+				isEnabled,
+				roomIds,
+				customMessage,
+				startDate: getISODateStringFromDateAndTime(startDateString, startTimeString),
+				endDate: getISODateStringFromDateAndTime(endDateString, endTimeString),
+			}),
+			[
+				roomIds,
+				customMessage,
+				isEnabled,
+				startDateString,
+				startTimeString,
+				endDateString,
+				endTimeString,
+			],
+		),
+	);
+
+	const handleSaveChanges = useCallback(async () => {
+		const result = await toggleOutOfOffice();
+		if (result && result.success === true) {
+			dispatchToastMessage({ type: 'success', message: result.message });
+			commit();
+		}
+	}, [commit, dispatchToastMessage, toggleOutOfOffice]);
+
+	const { value: { update: subscribedRooms = [] } = { update: [] } }: IEndpointSubscriptionsGet =
+		useEndpointData('subscriptions.get' as any);
+
+	const roomOptions: Array<[string, string]> = useMemo(
+		() => subscribedRooms.filter((s) => s.t !== 'd').map((s) => [s.rid, s.name]),
+		[subscribedRooms],
+	);
+
+	return (
+		<Page>
+			{/* @ts-ignore */}
+			<Page.Header title={t('Out Of Office')}>
+				<ButtonGroup>
+					<Button primary disabled={!hasUnsavedChanges} onClick={handleSaveChanges}>
+						{t('Save_changes')}
+					</Button>
+				</ButtonGroup>
+				{/* @ts-ignore */}
+			</Page.Header>
+			{/* @ts-ignore */}
+			<Page.ScrollableContentWithShadow>
+				<Box maxWidth='x800' w='full' alignSelf='center'>
+					<FieldGroup>
+						<Field>
+							<Box display='flex' justifyContent='flex-start' alignItems='center'>
+								<Field.Row>
+									<RadioButton checked={!isEnabled} onChange={(): void => handleIsEnabled(false)} />
+								</Field.Row>
+								<Field.Row>
+									<Box display='flex' flexDirection='column'>
+										<Field.Label>Disable Out Of Office</Field.Label>
```

Comment:
please use i18n for labels

Reply:
I thought I would add the translations at the end but it seems I forgot! 😅
Please give me some time, I will add them.